clustering/jgroups/extension/src/main/java/org/jboss/as/clustering/jgroups/subsystem/JGroupsSubsystemResourceDefinition.java

```diff
@@ -110,11 +116,31 @@ public void register(SubsystemRegistration parentRegistration) {
                 .addAttributes(Attribute.class)
                 .addCapabilities(model -> model.hasDefined(Attribute.DEFAULT_CHANNEL.getName()), CAPABILITIES.values())
                 .addCapabilities(model -> model.hasDefined(Attribute.DEFAULT_CHANNEL.getName()), capabilities)
+                // Tolerate duplicate add operations
+                .setAddOperationTransformation(ExecuteIfAbsentStepHandler::new)
                 ;
         ResourceServiceHandler handler = new JGroupsSubsystemServiceHandler();
         new SimpleResourceRegistrar(descriptor, handler).register(registration);
 
         new ChannelResourceDefinition().register(registration);
         new StackResourceDefinition().register(registration);
     }
+
+    private static class ExecuteIfAbsentStepHandler implements OperationStepHandler {
+
+        private final OperationStepHandler handler;
+
+        ExecuteIfAbsentStepHandler(OperationStepHandler handler) {
+            this.handler = handler;
+        }
+
+        @Override
+        public void execute(OperationContext context, ModelNode operation) throws OperationFailedException {
+            PathAddress address = context.getCurrentAddress();
+            Resource parent = context.readResourceFromRoot(address.getParent(), false);
+            if (!parent.hasChild(address.getLastElement())) {
```

Comment:
There is a still race between L140 and L142.

Reply:
@bstansberry Do you mean that operations generated by 2 subsystem parsers will execute concurrently?

Reply:
@pferraro Yes, that is part of what parallel boot does. Operations are grouped by subsystem and the groups are executed concurrently. Different subsystems updating each other is not allowed. I must not have noticed this when reviewing #6736.

TBH this is probably good enough. The InfinispanSubsystemXMLReader logic has been there since 2014 and apparently wasn't a big problem. Your fix makes it even less of one. Over time parsing documents using these is getting less and less common, so the whole thing seems like an edge case. But, your marking the JIRA as Critical makes me wonder if I'm not understanding something.

Reply:
Yeah - the problematic logic in InfinispanSubsystemXMLReader only exists to support XML schemas from AS 7.x.  Hopefully, we can remove this soon?
I sent you a DM explaining the reason for the critical priority.